### PR TITLE
Feature favorites storage

### PR DESCRIPTION
**Pull request by demaceo**
#### What does  this PR do?
- Stores favorited songs in local storage
- Adds musical note animation

#### Any background context you want to provide?
- ~~Currently, the 'View Favorites' does pull from local storage successfully & remain there on page refresh. HOWEVER, if a user refreshes the page & favorites a new song, only that song will appear in the 'View Favorites'. If we could do something like `window.onload(setFavoriteSongs(localStorage.getItem('favorites')` that would be groovy, but 1) thats not proper syntax & 2) even if it was TypeScript would probably throw a fit about it.~~
- The styling & animation are completely optional. We can get rid of it if it's too distracting or not thematic.
- I noticed that sometimes the 'Get Songs' button in the Form requires a double click. Though, on the first click one set of results are given, & then on the second click a completely new set of results are given.

### CODE REVIEW

**Review by brabbuss — CHANGES_REQUESTED** [review]
Great work! One issue I'm running into is when I click on favorites, I get this error:

Screen Shot 2021-01-10 at 10 41 30 PM

@demaceo @bruce-gordon What do you think is going on? Maybe the way the default is set if no local storage item? I don't want to mess with it bc this is your sweet baby.

**Review by bruce-gordon — COMMENTED** [review]
Favorites display is one song behind the actual number of favorites.

**Review by bruce-gordon — APPROVED** [review]

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -23,17 +26,30 @@ function App() {
...
+    }
   }
 
   const removeFavorite = (id: string) => {
     const favorites = favoriteSongs.filter((song:ISongResults) => song.id !== id) as any;
     setFavoriteSongs(favorites);
+    setLocalStorage(favorites);
   }
 
+  let storedFavs: any = useLocalStorage('favorites');
```
Comment (brabbuss) [inline src/components/App/App.tsx:47]: That seems to be the ticket

**Review by brabbuss — APPROVED** [review]
Glad we worked through it this AM

**Pull request merged**